crates/subspace-farmer/src/commitments/commitment_databases.rs

```diff
@@ -100,27 +70,29 @@ impl CommitmentDatabases {
                     true
                 }
                 CommitmentStatus::Created => false,
-            })
-            .next()
-            .is_some()
-        {
-            commitment_databases.persist_metadata_cache()?;
-        }
-
-        // Open databases that were fully created during previous run
-        for salt in commitment_databases.metadata_cache.keys() {
-            let db = DB::open(&Options::default(), base_directory.join(hex::encode(salt)))
-                .map_err(CommitmentError::CommitmentDb)?;
-            commitment_databases.databases.put(
-                *salt,
-                Arc::new(DbEntry {
-                    salt: *salt,
-                    db: Mutex::new(Some(Arc::new(db))),
-                }),
-            );
-        }
+            });
 
-        Ok::<_, CommitmentError>(commitment_databases)
+            // Open databases that were fully created during previous run
+            for salt in metadata.keys() {
+                let db = DB::open(&Options::default(), base_directory.join(hex::encode(salt)))
+                    .map_err(CommitmentError::CommitmentDb)?;
+                databases.put(
+                    *salt,
+                    Arc::new(DbEntry {
+                        salt: *salt,
+                        db: Mutex::new(Some(Arc::new(db))),
+                    }),
+                );
+            }
+
+            Ok(())
+        })?;
+
+        Ok::<_, CommitmentError>(CommitmentDatabases {
```

Comment:
Isn't error here inferred :thinking: ?

Reply:
Indeed, probably left from earlier code from which this one is derived